src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonParameterInfo.cs

```diff
@@ -59,11 +59,10 @@ public JsonClassInfo RuntimeClassInfo
         public bool ShouldDeserialize { get; private set; }
 
         public virtual void Initialize(
-            string matchingPropertyName,
             Type declaredPropertyType,
             Type runtimePropertyType,
             ParameterInfo parameterInfo,
-            JsonConverter converter,
```

Comment:
Hmm so `converter` wasn't used I assume.

Reply:
`converter` was used, but we now fetch it from `matchingProperty.ConverterBase`.